### PR TITLE
LPAL-588 add content-security-policy and x-content-security-policy headers including moving all inlined js into separate files to make this work

### PR DESCRIPTION
**Pull request by nickdavis2001**
## Purpose

_LPAL-588 add content-security-policy and x-content-security-policy headers, including moving all inlined js into seperate files_

## Approach

_Add headers to PHP, and to nginx in case PHP ever loses its headers. Also check in cypress test.  2 weird things to note- (1) had to do formatting that looks inconsistent to me on the js to keep prettier happy. (2) a semi-colon at the end is expected for content-security-policy but not for x-content-security-policy_

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added mandatory tags to terraformed resources, where possible
* [ ] If I have a new OPG component dependency, I have updated the `metadata.json` with the repo location.
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/module/Application/src/Controller/AbstractBaseController.php [hunk 1]
```diff
@@ -217,9 +217,18 @@ public function onDispatch(\Laminas\Mvc\MvcEvent $e)
...
 
+        $securityPolicy = "default-src 'self'";
+
         $response->getHeaders()->addHeaders([
             'X-Content-Type-Options' => 'nosniff',
-            'Referrer-Policy' => 'strict-origin-when-cross-origin'
+            'Referrer-Policy' => 'strict-origin-when-cross-origin',
+        ]);
+
+        $response->getHeaders()->addHeaders([
```
Comment (townxelliot) [inline service-front/module/Application/src/Controller/AbstractBaseController.php:227]: Could these lines be merged into the previous addHeaders() call?

File: service-front/module/Application/src/Controller/AbstractBaseController.php [hunk 2]
```diff
@@ -217,9 +217,18 @@ public function onDispatch(\Laminas\Mvc\MvcEvent $e)
...
+            'Content-Security-Policy' => $securityPolicy
+        ]);
+        $response->getHeaders()->addHeaders([
+            'X-Content-Security-Policy' => $securityPolicy
```
Comment (townxelliot) [inline service-front/module/Application/src/Controller/AbstractBaseController.php:231]: ditto

**Review by townxelliot — APPROVED** [review]

File: service-front/module/Application/src/Controller/AbstractBaseController.php (hunk 2 above)
Comment (nickdavis2001) [reply service-front/module/Application/src/Controller/AbstractBaseController.php]: Done, was temporary measure while debugging.

**Review by townxelliot — CHANGES_REQUESTED** [review]

File: service-front/assets/js/main.js [hunk 3]
```diff
@@ -15,3 +15,20 @@ showHideContent.init();
 
 // Remove the no-js class
 $('body').removeClass('no-js');
+
+var dashboard = function () {
```
[the remark this answers is not in the document]
Comment (nickdavis2001) [reply service-front/assets/js/main.js:19]: No longer needed now, so removed

**Review by townxelliot — APPROVED** [review]

**Review by townxelliot — APPROVED** [review]

**Pull request merged**